File: scripts/highlight_residues.js
const { exec } = require('child_process');
const fs = require('fs');
const path = require('path');

// 配置参数
const pdbFile = process.argv[2] || 'BAB64565.3_model01.pdb';
const outputImage = process.argv[3] || 'GH7_positive_sites.png';
const residues = [58, 184, 225, 312]; // 正选择位点

// 检查输入文件是否存在
if (!fs.existsSync(pdbFile)) {
    console.error(`错误: 文件 ${pdbFile} 不存在`);
    process.exit(1);
}

console.log(`正在处理 ${pdbFile}...`);

// 构建 iCn3D 命令序列
// iCn3D 命令行工具通过 'commands' 参数传递一系列操作
const commands = [
    'style sphere',                              // 球体显示
    'color grey',                                // 整体灰色
    `select residue ${residues.join(',')}`,       // 选择残基
    'color red',                                 // 选中的残基设为红色
    'style stick',                               // 选中的残基显示为棍状
    'view',                                      // 调整视角
    `save PNG ${outputImage}`,                   // 保存PNG
].join(';');

// 注意: iCn3D 的 Node.js 包需要在项目中安装
// 以下是使用 icn3d npm 包的示例代码（需要先 npm install icn3d）
console.log('请确保已安装 icn3d: npm install icn3d');

// 实际调用 icn3d 的示例代码 (需要在 icn3d 环境中运行)
/*
const icn3d = require('icn3d');
const ui = new icn3d.iCn3DUI({});
ui.loadFile(pdbFile).then(() => {
    ui.executeCommands(commands);
    console.log(`图片已保存为 ${outputImage}`);
});
*/

console.log(`命令序列: ${commands}`);
console.log('提示: 本脚本需要安装 Node.js 和 icn3d 包。');
console.log('如果尚未安装，请运行: npm install icn3d jquery jsdom');

File: scripts/highlight_residues.py
import py3Dmol

pdb_file = "model_01.pdb"
with open(pdb_file, 'r') as f:
    pdb_data = f.read()

view = py3Dmol.view(width=1200, height=800)
view.addModel(pdb_data, 'pdb')
view.setStyle({'cartoon': {'color': 'gray'}})

residues = [58, 184, 225, 312]
view.addStyle({'resi': residues}, {'stick': {'color': 'red'}, 'sphere': {'color': 'red', 'radius': 0.3}})

view.zoomTo()

html_file = "GH7_positive_sites.html"
with open(html_file, 'w') as f:
    f.write(view._make_html())

print(f"已生成 {html_file}，请用浏览器打开该文件并截图保存。")
